### PR TITLE
Vault - Guard check for add reward

### PR DESCRIPTION
**Pull request by amandeeprehal-lab49**
Vault - Guard check for add reward

### CODE REVIEW

File: e2e-test/business/Vault.ts [hunk 1]
```diff
@@ -32,13 +33,15 @@ export default class Vault extends Base {
   initialize = async (
     stakingToken: TokenId,
     lockingPeriod: BigNumber | number,
+    systemRoleBasedAccessAddress: string,
```
Comment (shivamgarglab49) [inline e2e-test/business/Vault.ts:36]: We having one getter inside Base i.e getSystemBasedRoleAccessContractAddress() to get the systemRoleBasedAccessAddress,

Please check if we can use it.

File: integrationTest/incentives/vault.ts [hunk 2]
```diff
@@ -46,7 +48,15 @@ const stake = async (vault: Vault) => {
 
 async function main() {
   const vault = new Vault();
-  await vault.initialize(STAKING_TOKEN, LOCKING_PERIOD_IN_SECONDS);
+  const systemRoleBasedAccess = new SystemRoleBasedAccess();
```
Comment (shivamgarglab49) [inline integrationTest/incentives/vault.ts:51]: May be not needed if we call it internally inside initialize call

File: test/vault-test.ts [hunk 3]
```diff
@@ -381,6 +388,18 @@ describe("Vault Tests", function () {
   });
 
   describe("Add rewards tests", function () {
+    it.only("When non-reward-user try to add reward then should fail", async function () {
```
Comment (shivamgarglab49) [inline test/vault-test.ts:391]: please remove 'only'

File: e2e-test/business/Vault.ts [hunk 4]
```diff
@@ -9,6 +9,7 @@ import {
   TokenId,
   AccountId,
   ContractFunctionParameters,
+  ContractId,
```
Comment (shivamgarglab49) [inline e2e-test/business/Vault.ts:12]: unused import

File: integrationTest/incentives/vault.ts (hunk 2 above)
Comment (amandeeprehal-lab49) [reply integrationTest/incentives/vault.ts:51]: right, changed

File: e2e-test/business/Vault.ts (hunk 1 above)
Comment (amandeeprehal-lab49) [reply e2e-test/business/Vault.ts:36]: Done

File: e2e-test/business/Vault.ts (hunk 4 above)
Comment (amandeeprehal-lab49) [reply e2e-test/business/Vault.ts:12]: Done

File: test/vault-test.ts (hunk 3 above)
Comment (amandeeprehal-lab49) [reply test/vault-test.ts]: done

**Review by shivamgarglab49 — APPROVED** [review]

**Pull request merged**